### ISSUE TITLE
Regional GKE clusters arent listed using current Clusters.List() implementation

### ISSUE THREAD

**Issue by Alphasite** [genevieve/leftovers#108]
As per the title the current approach of listing cluster per zone misses regional clusters, so we should be using [this API](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters/list) which allows listing clusters across all regions (and as a plus this should give a nice perf boost).

I have the relevant changes in my own fork: https://github.com/Alphasite/leftovers (and some cleanup to bump ginkgo and remove the vendor directory since newer go tooling doesn't play extremely well with it).

**Comment by genevieve** [comment]
Hey @Alphasite! I was in the process of doing those upgrades and cleanup but if you'd like to make an MR for the changes, please do!

**Comment by Alphasite** [comment]
Would you prefer i do them as separate PRs or is 1 big one ok?

**Comment by Alphasite** [comment]
Sure, thanks! And i'm glad this project is still alive, i appreciate the responsiveness! I was almost tempted to ping @rowanjacobs and ask 😅.

**Comment by Alphasite** [comment]
Done! Created #109 #110 and #111 thank you!

**Comment by Alphasite** [comment]
(If there is anything we can do to help, please let me know)

**Comment by Alphasite** [comment]
Resolved in #111 (Is the convention to leave an issue open until its fix is merged or released?)

**Comment by genevieve** [comment]
I have some ci work to do to move from concourse to github actions and then I plan to use that to cut a new release. Are you able to use the master branch for your purposes for now?

**Comment by Alphasite** [comment]
Yeah, it’s not an issue at all, I’m just using a build off main and it’s
working great, thank you for looking at it so quickly!
